lib/mariaex/messages.ex

```diff
@@ -214,11 +216,15 @@ defmodule Mariaex.Messages do
   defp encode_param(int) when is_integer(int),
     do: {0, :field_type_longlong, << int :: 64-little >>}
   defp encode_param(float) when is_float(float),
-    do: {0, :field_type_newdecimal, << float :: 64-little-float >>}
+    do: {0, :field_type_double, << float :: 64-little-float >>}
   defp encode_param(true),
     do: {0, :field_type_tiny, << 01 >>}
   defp encode_param(false),
     do: {0, :field_type_tiny, << 00 >>}
+  defp encode_param(%Decimal{} = value) do
+    bin = Decimal.to_string(value, :normal)
+    {0, :field_type_newdecimal, << to_length_encoded_integer(byte_size(bin)) :: binary, bin :: binary >>}
```

Comment:
I'm not sure. Unfortunately MySQL doesn't provide the documentation for this protocol.

http://dev.mysql.com/doc/internals/en/binary-protocol-value.html#packet-ProtocolBinary::MYSQL_TYPE_DECIMAL

But according this link http://dev.mysql.com/doc/refman/5.0/en/fixed-point-types.html the maximum number of digits is 65.